src/utils/settings/data.js

```diff
@@ -0,0 +1,348 @@
+  }
+  return format.displayBgValue(bgValue, units);
+}
+
+/**
+ * getStarts
+ * @private
+ * @param  {Array} timedData array with time based data
+ *
+ * @return {Array}           array of start times in milliseconds
+ */
+function getStarts(timedData) {
+  return _.map(timedData, 'start');
+}
+
+/**
+ * getTotalBasalRates
+ * @param  {Array} scheduleData  basal schedule data
+ *
+ * @return {String}              formatted total of basal rates
+ */
+export function getTotalBasalRates(scheduleData) {
+  const HOUR_IN_MILLISECONDS = 60 * 60 * 1000;
+  const DAY_IN_MILLISECONDS = 86400000;
+
+  let total = 0;
+  for (let i = scheduleData.length - 1; i >= 0; i--) {
+    const start = scheduleData[i].start;
+    let finish = DAY_IN_MILLISECONDS;
+    const next = i + 1;
+    if (next < scheduleData.length) {
+      finish = scheduleData[next].start;
+    }
+    const hrs = (finish - start) / HOUR_IN_MILLISECONDS;
+    total += (scheduleData[i].rate * hrs);
+  }
+  return format.displayDecimal(total, DISPLAY_PRECISION_PLACES);
+}
+
+/**
+ * getScheduleLabel
+ * @param  {String} scheduleName basal schedule name
+ * @param  {String} activeName   basal name active at upload timestamp
+ *
+ * @return {String}              formatted basal schedule label
+ */
+export function getScheduleLabel(scheduleName, activeName) {
+  if (scheduleName === activeName) {
+    return `${scheduleName} (Active at upload)`;
+  }
+  return scheduleName;
+}
+
+/**
+ * getScheduleNames
+ * @param  {Object} settingsData object with basal schedule properties
+ *
+ * @return {Array}               array of basal schedule names
+ */
+export function getScheduleNames(settingsData) {
+  return _.keysIn(settingsData);
+}
+
+/**
+ * getTimedSchedules
+ * @param  {Array} settingsData array of basal schedules
+ *
+ * @return {Array}              array of {name, position} basal objects
+ */
+export function getTimedSchedules(settingsData) {
+  const names = _.map(settingsData, 'name');
+  const schedules = [];
+  for (let i = names.length - 1; i >= 0; i--) {
+    schedules.push({ name: names[i], position: i });
+  }
+  return schedules;
+}
+
+/**
+ * getDeviceMeta
+ * @param  {Object} settingsData all settings data
+ * @param  {Object} timePrefs    timezone preferences object
+ *
+ * @return {Object}              filtered meta data
+ */
+export function getDeviceMeta(settingsData, timePrefs) {
+  let uploadedTime;
+  if (timePrefs.timezoneAware) {
+    if (settingsData.time) {
+      uploadedTime = datetime.formatDisplayDate(moment(settingsData.time).valueOf(), timePrefs);
+    } else {
+      uploadedTime = 'unknown';
+    }
+  } else {
+    if (settingsData.deviceTime) {
+      uploadedTime = datetime.formatDisplayDate(moment(settingsData.deviceTime).valueOf(),
+        timePrefs);
+    } else {
+      uploadedTime = 'unknown';
+    }
+  }
+  return {
+    name: settingsData.deviceId || 'unknown',
+    schedule: settingsData.activeSchedule || 'unknown',
+    uploaded: uploadedTime,
```

Comment:
For consistency, could we just delay the assignment of 'unknown' for `uploadedTime` to here?